oak-search/src/test/java/org/apache/jackrabbit/oak/plugins/index/IndexQueryCommonTest.java

```diff
@@ -167,6 +177,26 @@ public void descendantTestWithIndexTag() throws Exception {
         });
     }
 
+    @Test
+    public void descendantTestWithIndexTagExplain() throws Exception {
+        Tree test = root.getTree("/").addChild("test");
+        test.addChild("a");
+        test.addChild("b");
+        root.commit();
+
+        String query = "explain select [jcr:path] from [nt:base] where isdescendantnode('/test') option (index tag x)";
+        assertEventually(getAssertionForExplain(query, Query.JCR_SQL2, getExplainValueForDescendantTestWithIndexTagExplain(), true));
```

Comment:
Unrelated to this PR, a general comment.
I wonder if there is a better way to check that the index was created. The way we are using assertEventually(), is testing for two conditions: 

1) Index was created - Must retry until this is true by polling.
2) the test condition passes - If it fails, the test can be aborted immediately. 

But currently we are retrying for failures in 1 (correct) and 2 (not needed), which will slow down the tests in case of failures of the test condition, as it will keep retrying a test that will not pass until reaching the retry limit. 

Maybe there is a way of separating the two conditions, with a generic polling test to check if the index was updated, separate from the condition under test? The slowdown occurs only when there are failures, but this is the common case when we are developing, so fixing this issue would be a nice quality of life improvement for us while we are in the edit-compile-test loop. And it would also make the intention of the code more clean, by separating logic from the test framework from the logic specific to the current test.